src/test/osd/scrubber_generators.cc

```diff
@@ -0,0 +1,164 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+#include "test/osd/scrubber_generators.h"
+
+using namespace ScrubGenerator;
+
+// ref: PGLogTestRebuildMissing()
+bufferptr create_object_info(const ScrubGenerator::RealObj& objver)
+{
+  object_info_t oi{};
+  oi.soid = objver.ghobj.hobj;
+  oi.version = eversion_t(objver.ghobj.generation, 0);
+  oi.size = objver.data.size;
+
+  bufferlist bl;
+  oi.encode(bl,
+	    0 /*get_osdmap()->get_features(CEPH_ENTITY_TYPE_OSD, nullptr)*/);
+  bufferptr bp(bl.c_str(), bl.length());
+  return bp;
+}
+
+std::pair<bufferptr, std::vector<snapid_t>> create_object_snapset(
+  const ScrubGenerator::RealObj& robj,
+  const SnapsetMockData* snapset_mock_data)
+{
+  if (!snapset_mock_data) {
+    return {bufferptr(), {}};
+  }
+  /// \todo fill in missing version/osd details from the robj
+  auto sns = snapset_mock_data->make_snapset();
+  bufferlist bl;
+  encode(sns, bl);
+  bufferptr bp = bufferptr(bl.c_str(), bl.length());
+
+  // extract the set of object snaps
+  return {bp, sns.snaps};
```

Comment:
Nit: Can't we return `{bl, sns.snaps};`? 
line 73: `ret.smobj.attrs[SS_ATTR] = std::move(bl);`

Reply:
The 'bl' is a buffer-list, not a buffer pointer. What am I missing?